services/apps/src/apps/service/apps/de/categorization.clj

```diff
@@ -5,8 +5,42 @@
         [kameleon.entities]
         [apps.validation]
         [slingshot.slingshot :only [throw+]])
-  (:require [apps.persistence.app-metadata :as ap]
-            [apps.service.apps.de.validation :as av]))
+  (:require [apps.clients.permissions :as perms-client]
+            [apps.persistence.app-metadata :as ap]
+            [apps.persistence.categories :as db-categories]
+            [apps.service.apps.de.validation :as av]
+            [apps.util.config :as config]
+            [metadata-client.core :as metadata-client]))
+
+(defn get-active-hierarchy-version
+  []
+  (let [version (db-categories/get-active-hierarchy-version)]
+    (when-not version
+      (throw+ {:type  :clojure-commons.exception/not-found
+               :error "An app hierarchy version has not been set."}))
+    version))
+
+(defn validate-app-name-in-hierarchy-avus
+  [username app-id app-name avus]
+  (let [category-attrs    (set (config/workspace-metadata-category-attrs))
+        hierarchy-avus    (->> avus
+                               (filter #(contains? category-attrs (:attr %)))
+                               (map #(select-keys % [:attr :value])))
+        other-app-ids     (disj (set (keys (perms-client/load-app-permissions username))) app-id)
```

Comment:
I wonder if it would be better to just pass the app's ID along with its name to the [database query](https://github.com/cyverse/DE/pull/299/files#diff-e8fd70ff48696b547e301e34fb2370c3R854), and let SQL exclude this ID?
Maybe with something like
```sql
WHERE id IN ($app_id_set) AND id != $app_id
```